### PR TITLE
Ch accounts tests #139551323

### PR DESCRIPTION
**Pull request by daisyndungu**
Add tests in Accounts.

### CODE REVIEW

File: hc/accounts/tests/test_login.py [hunk 1]
```diff
@@ -13,20 +13,22 @@ def test_it_sends_link(self):
         session = self.client.session
         session["welcome_code"] = str(check.code)
         session.save()
-
+        self.assertEqual(Profile.objects.count(), 0)
+        # count before
```
Comment (osule) [inline hc/accounts/tests/test_login.py]: Can we have this comment inlined with the previous line?

File: hc/accounts/tests/test_login.py [hunk 2]
```diff
@@ -13,20 +13,22 @@ def test_it_sends_link(self):
...
 
         ### Assert that a user was created
-
+        self.assertEqual(Profile.objects.count(), 1)
         # And email sent
         self.assertEqual(len(mail.outbox), 1)
         self.assertEqual(mail.outbox[0].subject, 'Log in to healthchecks.io')
         ### Assert contents of the email body
-
+        assert "To log into healthchecks.io" in mail.outbox[0].body
```
Comment (osule) [inline hc/accounts/tests/test_login.py]: We should take advantage of assertion methods available in the django TestCase class. https://docs.djangoproject.com/en/1.10/topics/testing/tools/#django.test.SimpleTestCase.assertContains

File: hc/accounts/tests/test_profile.py [hunk 3]
```diff
@@ -18,8 +18,10 @@ def test_it_sends_set_password_link(self):
         self.alice.profile.refresh_from_db()
         token = self.alice.profile.token
         ### Assert that the token is set
-
+        self.assertEqual(self.alice.profile.token, token)
         ### Assert that the email was sent and check email content
+        self.assertEqual(len(mail.outbox),1)
+        assert "link" in mail.outbox[0].body
```
Comment (osule) [inline hc/accounts/tests/test_profile.py]: Please use assertContains method available in Django SimpleTestCase class.
It provides metadata about the test on failure.

I'm well aware of other testing frameworks promoting use of assert stmts such as py.test but I think that if we are using Unittest here we should take advantage of the good things that it provides us with.
https://docs.python.org/2/reference/simple_stmts.html#the-assert-statement

Comment (osule) [reply hc/accounts/tests/test_profile.py]: Oh, and there is the issue of running python with the -O flag in an optimized state which loses assert stmts.

File: hc/accounts/tests/test_profile.py [hunk 4]
```diff
@@ -45,6 +52,11 @@ def test_it_adds_team_member(self):
         self.assertTrue("frank@example.org" in member_emails)
 
         ###Assert that the email was sent and check email content
+        self.assertEqual(len(mail.outbox), 1)
+        ##checks the contents of the email subject
+        self.assertEqual(mail.outbox[0].subject, 'You have been invited to join alice@example.org on healthchecks.io')
+        ##checks the contents of the email body
+        assert "Hello,\n\nalice@example.org invites you" in mail.outbox[0].body
```
Comment (osule) [inline hc/accounts/tests/test_profile.py]: Please see previous review comments

File: hc/accounts/tests/test_profile.py [hunk 5]
```diff
@@ -108,3 +120,8 @@ def test_it_shows_badges(self):
         self.assertNotContains(r, "bobs-tag.svg")
 
     ### Test it creates and revokes API key
+    def test_it_creates_and_revokes_api_key(self):
```
Comment (osule) [inline hc/accounts/tests/test_profile.py]: We should be testing out this https://github.com/andela/healthchecks_akira/blob/df137bfdad57cb53bcf31bdc92056cb4f316f921/hc/accounts/views.py#L150

Hit the endpoint for that view with a POST request to create an API token, test that the token is valid. Make another POST request to revoke the token and test that the token is invalid.

File: hc/accounts/tests/test_switch_team.py [hunk 6]
```diff
@@ -15,17 +15,18 @@ def test_it_switches(self):
 
         ### Assert the contents of r
 
-
+        assert r.status_code == 200
```
Comment (osule) [inline hc/accounts/tests/test_switch_team.py]: Please make use of `assertEqual` instance method of the Django TestCase class

File: hc/accounts/tests/test_switch_team.py [hunk 7]
```diff
@@ -15,17 +15,18 @@ def test_it_switches(self):
...
+        assert r.status_code == 200 
     def test_it_checks_team_membership(self):
         self.client.login(username="charlie@example.org", password="password")
 
         url = "/accounts/switch_team/%s/" % self.alice.username
         r = self.client.get(url)
-        ### Assert the expected error code
-
+        ### Assert the expected error code(Forbidden access)
+        assert r.status_code == 403
```
Comment (osule) [inline hc/accounts/tests/test_switch_team.py]: Please use the assertEqual instance method of the Django TestCase class

File: hc/accounts/tests/test_switch_team.py [hunk 8]
```diff
@@ -15,17 +15,18 @@ def test_it_switches(self):
...
     def test_it_switches_to_own_team(self):
         self.client.login(username="alice@example.org", password="password")
 
         url = "/accounts/switch_team/%s/" % self.alice.username
         r = self.client.get(url, follow=True)
-        ### Assert the expected error code
+        ### Assert the expected error code(Accepted access)
+        assert r.status_code == 200
```
Comment (osule) [inline hc/accounts/tests/test_switch_team.py]: Please see previous review comment about this

File: hc/settings.py [hunk 9]
```diff
@@ -20,10 +20,14 @@
 HOST = "localhost"
 SECRET_KEY = "---"
 DEBUG = True
-ALLOWED_HOSTS = []
-DEFAULT_FROM_EMAIL = 'healthchecks@example.org'
+# ALLOWED_HOSTS = []
```
Comment (osule) [inline hc/settings.py]: What are your reasons for commenting this out?

File: hc/settings.py [hunk 10]
```diff
@@ -20,10 +20,14 @@
...
+# DEFAULT_FROM_EMAIL = 'healthchecks@example.org'
 USE_PAYMENTS = False
-
+EMAIL_HOST = "smtp.gmail.com"
```
Comment (osule) [inline hc/settings.py]: We shouldn't check this in.

**Review by osule — CHANGES_REQUESTED** [review]
Please see all review comments and attend to them accordingly.

File: hc/settings.py (hunk 9 above)
Comment (daisyndungu) [reply hc/settings.py]: I did this in order to get an actual email in my mailbox

File: hc/settings.py (hunk 10 above)
Comment (daisyndungu) [reply hc/settings.py]: I did this in order to get an actual email in my mailbox. I was using my gmail account but I'll change this back to the default settings.

File: hc/settings.py [hunk 11]
```diff
@@ -83,20 +78,34 @@
 
 # Default database engine is SQLite. So one can just check out code,
 # install requirements.txt and do manage.py runserver and it works
-DATABASES = {}
-
-DATABASES['default'] = dj_database_url.parse('sqlite:///{BASE_DIR}/hc.sqlite'.format(BASE_DIR=BASE_DIR), conn_max_age=600)
+DATABASES = {
```
Comment (osule) [inline hc/settings.py]: We shouldn't revert back the changes to the settings file.

File: hc/accounts/tests/test_profile.py (hunk 5 above)
Comment (osule) [reply hc/accounts/tests/test_profile.py]: @Andela-Didacus @daisyndungu We aren't exercising the API in this test. Please see my previous comment on this.

**Review by osule — CHANGES_REQUESTED** [review]

**Review by daisyndungu — COMMENTED** [review]
Changes made sucessfuly

**Review by osule — APPROVED** [review]
LGTM 👍

**Pull request merged**